### PR TITLE
Support retry times

### LINKED ISSUE #27: Need to specify the retry times of the MPI Job
Current the launcher Job's backoffLimit is 6, and it can't be modified the by end user. I suggest that the end user should be able to modify it.

And I think it can be 0 by default.

### PR DESCRIPTION
**Pull request by cheyang**
Fix #27

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/mpi-operator/28)

### CODE REVIEW

**Comment by cheyang** [comment]
Thanks for your suggestions! @everpeace

File: pkg/controllers/mpi_job_controller.go [hunk 1]
```diff
@@ -915,6 +915,12 @@ func newWorker(mpiJob *kubeflow.MPIJob, desiredReplicas int32, gpus int) *appsv1
 		},
 	})
 
+	// set default BackoofLimit
+	if mpiJob.Spec.BackoffLimit == nil {
+		mpiJob.Spec.BackoffLimit = new(int32)
+		*mpiJob.Spec.BackoffLimit = 0
```
Comment (rongou) [inline pkg/controllers/mpi_job_controller.go]: Perhaps set this to 6 to be consistent with Kubernetes `Jobs`?

**Comment by cheyang** [comment]
@rongou , PTAL.

**Pull request merged**